src/tasks/dump.ts

```diff
@@ -0,0 +1,567 @@
+    formatInstruction(inst, toToken),
+  );
+  const orderWithoutAllowances = [
+    "fromAddress",
+    "fromSymbol",
+    "receivedAmount",
+    "balance",
+    "feePercent",
+  ] as const;
+  const order = needsAnyAllowance
+    ? ([
+        ...orderWithoutAllowances.slice(0, 2),
+        "needsAllowance",
+        ...orderWithoutAllowances.slice(2),
+      ] as const)
+    : orderWithoutAllowances;
+  const header = {
+    fromAddress: "token address",
+    fromSymbol: "symbol",
+    balance: "dumped amount",
+    feePercent: "fee %",
+    receivedAmount: `received amount${
+      toToken.symbol ? ` (${toToken.symbol})` : ""
+    }`,
+    needsAllowance: "needs allowance?",
+  };
+  console.log(chalk.bold("List of dumped tokens:"));
+  displayTable(header, formattedInstructions, order, {
+    balance: { align: Align.Right, maxWidth: 30 },
+    receivedAmount: { align: Align.Right, maxWidth: 30 },
+    fromSymbol: { maxWidth: 20 },
+  });
+  console.log();
+}
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+async function prompt(message: string) {
+  const response = await new Promise<string>((resolve) =>
+    rl.question(`${message} (y/N) `, (response) => resolve(response)),
+  );
+  return "y" === response.toLowerCase();
+}
+
+async function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+async function createAllowances(
+  allowances: TokenDetails[],
+  signer: Signer,
+  allowanceManager: string,
+) {
+  if (allowances.length !== 0) {
+    for (const token of allowances) {
+      console.log(
+        `Approving allowance manager to trade token ${displayName(token)}...`,
+      );
+      const result: ContractTransaction = await token.contract
+        .connect(signer)
+        .approve(allowanceManager, constants.MaxUint256);
+      await result.wait();
+    }
+  }
+}
+
+async function createOrders(
+  instructions: DumpInstruction[],
+  toToken: TokenDetails | NativeToken,
+  signer: Signer,
+  receiver: string,
+  hasCustomReceiver: boolean,
+  domainSeparator: TypedDataDomain,
+  validity: number,
+  network: SupportedNetwork,
+  ethers: HardhatRuntimeEnvironment["ethers"],
+) {
+  const blockTimestamp = (await ethers.provider.getBlock("latest")).timestamp;
+  // Check that the local time is consistent with that of the blockchain
+  // to avoid signing orders that are valid for too long
+  const now = Math.floor(Date.now() / 1000);
+  if (Math.abs(now - blockTimestamp) > MAX_LATEST_BLOCK_DELAY_SECONDS) {
+    throw new Error("Blockchain time is not consistent with local time.");
+  }
+
+  for (const inst of instructions) {
+    const order: Order = {
+      sellToken: inst.token.address,
+      buyToken: isNativeToken(toToken) ? BUY_ETH_ADDRESS : toToken.address,
+      sellAmount: inst.amount,
+      buyAmount: inst.receivedAmount,
+      feeAmount: inst.fee,
+      kind: OrderKind.SELL,
+      appData: "0x",
```

Comment:
nit: Let's use some real app data here (like keccak hash of "dump script" or the like, just so that we can identify it.